File: Template/Server/src/main/java/Service/IServiceUser.java
package Service;

import Exceptions.ServiceException;
import Model.User;

public interface IServiceUser {
    /**
     * Logins a user into the app.
     *
     * @param username Username passed by the user.
     * @param password Password passed by the user.
     * @return The user with the passed credentials.
     * @throws ServiceException If the user doesn't exist or the password is incorrect.
     */
    User login(String username, String password) throws ServiceException;
}

File: mpp-proiect-csharp-culbec/Server/Service/IServiceUser.cs


using Common.CommonUtils;
using Common.Exceptions;
using Model;

namespace Server.Service;

public interface IServiceUser
{
    /// <summary>
    ///     Logins a user into the app.
    /// </summary>
    /// <param name="username">Username of the user.</param>
    /// <param name="password">Password of the user.</param>
    /// <returns>The logged in user, if it exists.</returns>
    /// <exception cref="ServiceException">If the username couldn't be found or couldn't be retrieved.</exception>
    User Login(string username, string password);
}